src/main/java/com/growup/pms/status/repository/StatusQueryRepositoryImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.growup.pms.status.repository;
+
+import static com.growup.pms.status.domain.QStatus.status;
+
+import com.growup.pms.status.controller.dto.response.StatusResponse;
+import com.querydsl.core.types.Projections;
+import com.querydsl.core.types.dsl.BooleanExpression;
+import com.querydsl.jpa.impl.JPAQueryFactory;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Repository;
+
+@Slf4j
+@Repository
+@RequiredArgsConstructor
+public class StatusQueryRepositoryImpl implements StatusQueryRepository {
+
+    private final JPAQueryFactory queryFactory;
+
+    public List<StatusResponse> getAllStatusByProjectId(Long projectId) {
+        List<Long> ids = queryFactory
+                .select(status.id)
+                .from(status)
+                .where(
+                        isProjectId(projectId)
+                )
+                .fetch();
+
+        log.info("ids={}", ids);
+
+        if (ids == null || ids.isEmpty()) {
+            return new ArrayList<>();
+        }
+
+        return queryFactory
+                .select(Projections.constructor(StatusResponse.class,
+                        status.id,
+                        status.project.id,
+                        status.name,
+                        status.colorCode,
+                        status.sortOrder
+                ))
+                .from(status)
+                .where(
+                        status.id.in(ids)
+                )
+                .orderBy(status.sortOrder.asc())
+                .fetch();
+    }
```

Comment:
혹시 이 부분을 아래와 같이 수정하는 경우에는 문제가 있을지 여쭤봅니다! :thinking: 

```suggestion
    public List<StatusResponse> getAllStatusByProjectId(Long projectId) {
        return queryFactory
                .select(Projections.constructor(StatusResponse.class,
                        status.id,
                        status.project.id,
                        status.name,
                        status.colorCode,
                        status.sortOrder
                ))
                .from(status)
                .where(isProjectId(projectId))
                .orderBy(status.sortOrder.asc())
                .fetch();
    }
```

Reply:
제안하신대로 코드를 수정해도 동작상의 문제는 없을텐데, 커버링 인덱스를 적용해봤습니다!
혹시 불필요하다고 생각하셨을까요? 그렇다면 제안하신대로 수정을 하겠습니다 😊
일단 위에 `ids` 를 조회하는 과정에서도 `orderBy` 를 사용해야하는데 안해버렸네요 ㅋㅋ

Reply:
@longBright 아닙니다! 코드는 그대로 두셔도 괜찮을 것 같습니다 개인적으로 궁금한 부분이었는데 답을 듣고 이유를 알 수 있었습니다 :smile: 코드 자체에는 문제가 없어서 이대로 머지하겠습니다!

`fetch()` 같은 경우는 데이터가 없으면 빈 리스트를 반환하는 것으로 알고 있었어서 아래의 코드도 괜찮을 것 같습니다!

```java
if (ids.isEmpty()) {
    return Collections.emptyList(); // 이미 생성된 불변 리스트 반환
}
```

Reply:
넵 리뷰 감사합니다 ☺️ 이후 리팩토링 PR 에서 고려하고 반영하도록 하겠습니다!